quarkus/addons/knative/serving/runtime/src/main/java/org/kie/kogito/addons/quarkus/knative/serving/customfunctions/KnativeServiceRegistry.java

```diff
@@ -33,12 +35,25 @@ final class KnativeServiceRegistry {
 
     private final KnativeServiceDiscovery knativeServiceDiscovery;
 
+    private final VanillaKubernetesResourceDiscovery vanillaKubernetesResourceDiscovery;
+
     @Inject
-    KnativeServiceRegistry(KnativeServiceDiscovery knativeServiceDiscovery) {
+    KnativeServiceRegistry(KnativeServiceDiscovery knativeServiceDiscovery,
+            VanillaKubernetesResourceDiscovery vanillaKubernetesResourceDiscovery) {
         this.knativeServiceDiscovery = knativeServiceDiscovery;
+        this.vanillaKubernetesResourceDiscovery = vanillaKubernetesResourceDiscovery;
     }
 
     Optional<URI> getServiceAddress(String serviceName) {
-        return Optional.ofNullable(services.computeIfAbsent(serviceName, k -> knativeServiceDiscovery.query(KnativeServiceUri.parse(k)).orElse(null)));
+        String[] splitServiceName = serviceName.split("/");
+
+        switch (splitServiceName.length) {
+            case 1:
+                return Optional.ofNullable(services.computeIfAbsent(serviceName, k -> knativeServiceDiscovery.query(new KnativeServiceUri(null, serviceName)).orElse(null)));
+            case 2:
+                return Optional.ofNullable(services.computeIfAbsent(serviceName, k -> knativeServiceDiscovery.query(new KnativeServiceUri(splitServiceName[0], splitServiceName[1])).orElse(null)));
```

Comment:
@hbelmiro then lets do that 
```suggestion
       String[] splitServiceName = serviceName.split("/");
        final Function<String, Optional<URI>> function;
        switch (splitServiceName.length) {
            case 1:
                function = k -> knativeServiceDiscovery.query(new KnativeServiceUri(null, serviceName));
                break;
            case 2:
                function = k -> knativeServiceDiscovery.query(new KnativeServiceUri(splitServiceName[0], splitServiceName[1]));
                break;
            default:
                function = k -> vanillaKubernetesResourceDiscovery.query(VanillaKubernetesResourceUri.parse(k));
        }
        return Optional.ofNullable(services.computeIfAbsent(serviceName, k -> function.apply(k).orElse(null)));
```